### PR TITLE
stringzilla: add version 3.8.4

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **stringzilla/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
Wait for merged.
https://github.com/ashvardanian/StringZilla/pull/88

File: recipes/stringzilla/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -7,5 +7,9 @@ add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} PRIVATE stringzilla::stringzilla)
 target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
 if(stringzilla_VERSION VERSION_LESS 2.0)
-  target_compile_definitions(${PROJECT_NAME} PRIVATE STRINGZILLA_LESS_2_0)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE STRINGZILLA_API=1)
+elseif(stringzilla_VERSION VERSION_LESS 3.0)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE STRINGZILLA_API=2)
+else()
+  target_compile_definitions(${PROJECT_NAME} PRIVATE STRINGZILLA_API=3)
```
Comment (Ahajha) [inline recipes/stringzilla/all/test_package/CMakeLists.txt:14]: Could you put this variable directly in the CMakeToolchain, something like `tc.preprocessor_definitions["STRINGZILLA_API"] = Version(self.dependencies["stringzilla"].ref.version).major`?

Comment (toge) [reply recipes/stringzilla/all/test_package/CMakeLists.txt]: @Ahajha 
Thank you for your comment!
I fixed as you suggested.
Could you review it again?

File: recipes/stringzilla/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
-  // Find a substring like it's Waldo 🕵️‍♂️
-  size_t substring_position = strzl_naive_find_substr(haystack, needle);
+  auto substring_position = haystack.find(needle); // Or `rfind`
 
-  return EXIT_SUCCESS;
+  haystack.end() - haystack.begin() == haystack.size(); // Or `rbegin`, `rend`
+  haystack.find_first_of(" \w\t") == 4; // Or `find_last_of`, `find_first_not_of`, `find_last_not_of`
+  haystack.starts_with(needle) == true; // Or `ends_with`
+  haystack.remove_prefix(needle.size()); // Why is this operation in-place?!
```
Comment (Ahajha) [inline recipes/stringzilla/all/test_package/test_package.cpp:25]: Minor, for consistency add `return EXIT_SUCCESS;`.

File: recipes/stringzilla/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -49,4 +49,27 @@ int main(void) {
   return EXIT_SUCCESS;
 }
 
+#else
+
+int main(void) {
```
Comment (Ahajha) [inline recipes/stringzilla/all/test_package/test_package.cpp:54]: This test package is starting to get a little complicated but I don't know if there's anything to be done about that, as there's little duplication between the versions. Perhaps just have one function call per major version?

File: recipes/stringzilla/all/test_package/test_package.cpp (hunk 2 above)
Comment (toge) [reply recipes/stringzilla/all/test_package/test_package.cpp]: fixed.

File: recipes/stringzilla/all/test_package/test_package.cpp (hunk 3 above)
Comment (toge) [reply recipes/stringzilla/all/test_package/test_package.cpp]: fixed.

**Review by Ahajha — APPROVED** [review]
LGTM

File: recipes/stringzilla/all/test_package/test_package.cpp
Comment (valgur) [inline recipes/stringzilla/all/test_package/test_package.cpp:1]: I would split this into two separate .cpp files and handle the switch in CMake instead.

**Comment by toge** [comment]
@RubenRBS 
Could you review this PR?
The current version of StringZilla is 2.0.4 and StringZilla has been changed lots of API since 3.0.0.

**Review by RubenRBS — APPROVED** [review]
Thanks a lot for your patience while I circled back to this PR - The simplifications look great, thanks a lot! :)

**Review by perseoGI — APPROVED** [review]
LGTM!

**Pull request merged**